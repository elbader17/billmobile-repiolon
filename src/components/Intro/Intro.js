//Component that renders the introduction of the application
import React, { Component } from 'react';
import { Text, StyleSheet } from 'react-native';
import Swiper from 'react-native-swiper'; //https://github.com/leecade/react-native-swiper
import Screen1 from './Screens/Screen1';
import Screen2 from './Screens/Screen2';
import Screen3 from './Screens/Screen3';

class Intro extends Component {
  render() {
    return (
      //Configuration of component properties
      <Swiper
        showsButtons
        loop={false} 
        dotColor={gray} 
        style={styles.wrapper} 
        activeDotColor={white} 
        buttonWrapperStyle = {positionButtonNext}
        prevButton = {<Text style={styles.prev}></Text>} 
        nextButton = {<Text style={styles.styleButton}>Siguiente</Text>} 
      >
        <Screen1 />
        <Screen2 /> 
        <Screen3 /> 
      </Swiper>
    );
  }
}

var positionButtonNext = { top: 202, justifyContent: 'center'};
const white = '#fff';
const gray = '#C4C4C4';

const styles = StyleSheet.create({
  wrapper: {},
  styleButton: {
    width: 247, 
    height: 40, 
    padding: 10, 
    borderRadius: 2, 
    borderWidth: 1, 
    borderColor: white, 
    color: white,
    textAlign: 'center',
    fontFamily: 'Lato-Semibold', 
    fontSize: 14, 
  },
  prev: {}
});

export default Intro;