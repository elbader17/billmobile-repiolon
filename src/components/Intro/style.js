import { StyleSheet, Dimensions } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { COLORS } from '../../constants/colors';
import { FONTS } from '../../constants/fonts';

const styles = StyleSheet.create({
  silde: {
    flex: 1,
    backgroundColor: 'transparent',
    alignItems: 'center',
  },
  box1: {
    flex: 0.8,
    justifyContent: 'center',
  },
  box2: {
    flex: 0.2,
    justifyContent: 'center',
  },
  box3: {
    flex: 1,
    paddingHorizontal: 50
  },
  number: {
    fontFamily: FONTS.regular,
    fontSize: FONTS.size78,
    color: COLORS.white,
  },
  textTittle: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size18,
    color: COLORS.white,
    textAlign: 'center',
    width: wp('80%'),
    lineHeight: 32,
  },
  textDescription: {
    fontFamily: FONTS.latoLight,
    fontSize: FONTS.size16,
    color: COLORS.white,
    textAlign: 'center',
    width: wp('85%'),
    lineHeight: 21,
  },
  activeDotStyle: {
    backgroundColor: COLORS.white,
    width: 9,
    height: 9
  },
  dotStyle: {
    backgroundColor: COLORS.gray,
    width: 9,
    height: 9
  },
  button: {
    backgroundColor: 'transparent',
    borderColor: COLORS.white, 
    borderRadius: 30, 
    borderWidth: 1,
  },
  buttonText: {
    fontFamily: FONTS.latoRegular, 
    fontSize: FONTS.size16, 
    color: COLORS.white,
  }
});

export default styles;