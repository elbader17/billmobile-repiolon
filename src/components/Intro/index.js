import React from 'react';
import { View, Text, Image } from 'react-native';
import AppIntroSlider from 'react-native-app-intro-slider';
import {SLIDES_INTRO} from '../../constants/slides_intro';
import style from './style';

class Intro extends React.Component {
  _renderItem = (item) => {
    return (
      <View style={style.container} >
        <View style={style.silde}>
          <View style={style.box1}>
            <Image source={ item.image } style={ style.imageHeader } />  
          </View>
          <View style={style.box2}>
            <Text style={style.textTittle}>{item.text}</Text>
            <Text style={style.textDescription}>{item.description}</Text>
          </View>
        </View>
      </View>
    );
  }
  _onDone = () => {
    this.props.navigation.navigate('Login')
  }
  render() {
    return ( 
      <AppIntroSlider 
        renderItem={this._renderItem} 
        slides={SLIDES_INTRO} 
        onDone={this._onDone} 
        bottomButton
        buttonStyle={style.button}
        buttonTextStyle={style.buttonText}
        doneLabel='Listo'
        nextLabel='Siguiente'
        activeDotStyle={style.activeDotStyle}
        dotStyle={style.dotStyle}
      />
    )
  }
}

export default Intro;