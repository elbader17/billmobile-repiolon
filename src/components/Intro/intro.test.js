import React from 'react';
import { Alert } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from 'react-native-button';
import Intro from './Intro';

describe('Rendering Component Intro', () => {
  test('renders correctly', () => {
    const component = renderer.create(<Intro />).toJSON();
    expect(component).toMatchSnapshot();
  });
})

describe('Intro Button', () => { 
  it('button click should hide component', () => {
    const clickFn = jest.fn();
    const { alert } = Alert;
    Alert.alert = clickFn;
    const component = shallow(<Button onPress={clickFn}>Listo</Button>);
    component.simulate('press');
    Alert.alert = alert;
    expect(clickFn).toHaveBeenCalled();
  });

  it('encuentra button', () => {
    const wrapper = shallow(<Intro />);
    const button = wrapper.findWhere(node => node.prop('testID') === 'ready');
    expect(button).toBeTruthy();
  });
  
  it('introooo', () => {
    const clickFn = jest.fn();
    const { alert } = Alert;
    Alert.alert = clickFn;
    const wrapper = shallow(<Intro />);
    const button = wrapper.findWhere(node => node.prop('testID') === 'ready').first();
    button.dive().simulate('press');
    Alert.alert = alert;
    expect(clickFn).toHaveBeenCalled();
  })
});