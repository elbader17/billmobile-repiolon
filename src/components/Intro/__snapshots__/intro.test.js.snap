// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Component Intro renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    activeDotColor="#fff"
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDirection={true}
    autoplayTimeout={2.5}
    bounces={false}
    buttonWrapperStyle={
      Object {
        "justifyContent": "center",
        "top": 202,
      }
    }
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {},
      ]
    }
    contentOffset={
      Object {
        "x": 0,
      }
    }
    disableNextButton={false}
    dotColor="#C4C4C4"
    horizontal={true}
    index={0}
    loadMinimal={false}
    loadMinimalSize={1}
    loop={false}
    nextButton={
      <Text
        style={
          Object {
            "borderColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "color": "#fff",
            "fontFamily": undefined,
            "fontSize": undefined,
            "height": 40,
            "padding": 10,
            "textAlign": "center",
            "width": 247,
          }
        }
      >
        Siguiente
      </Text>
    }
    onIndexChanged={[Function]}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    prevButton={
      <Text
        style={Object {}}
      />
    }
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={true}
    showsHorizontalScrollIndicator={false}
    showsPagination={true}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3687d1",
              "height": 1334,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "bottom": 422,
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": 80,
                "lineHeight": 90,
                "position": "absolute",
              }
            }
          >
            1
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": 16,
                "height": 33,
                "lineHeight": 32,
                "position": "absolute",
                "textAlign": "center",
                "top": 287,
                "width": 289,
              }
            }
          >
            Facturar sin demoras
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": undefined,
                "height": 66,
                "lineHeight": 21,
                "position": "absolute",
                "textAlign": "center",
                "top": 332,
                "width": 289,
              }
            }
          >
            Como cuando lo era cuando lo hacias en papel, ágil para vos y para tus clientes.
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3687d1",
              "height": 1334,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "bottom": 422,
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": 80,
                "lineHeight": 90,
                "position": "absolute",
              }
            }
          >
            2
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": 16,
                "height": 33,
                "lineHeight": 32,
                "position": "absolute",
                "textAlign": "center",
                "top": 287,
                "width": 289,
              }
            }
          >
            Hacemos práctica la facturación
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": undefined,
                "height": 66,
                "lineHeight": 21,
                "position": "absolute",
                "textAlign": "center",
                "top": 332,
                "width": 289,
              }
            }
          >
            No tendrás que cargar la misma información dos veces. Nuestro sistema guarda y ayuda a hacer más ágil tu facturación.
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3687d1",
              "height": 1334,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "bottom": 422,
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": 80,
                "lineHeight": 90,
                "position": "absolute",
              }
            }
          >
            3
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": 16,
                "height": 33,
                "lineHeight": 32,
                "position": "absolute",
                "textAlign": "center",
                "top": 287,
                "width": 289,
              }
            }
          >
            Información segura
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#fff",
                "fontFamily": undefined,
                "fontSize": undefined,
                "height": 66,
                "lineHeight": 21,
                "position": "absolute",
                "textAlign": "center",
                "top": 332,
                "width": 289,
              }
            }
          >
            Nuestra política de privacidad es simple: No compartimos ninguna información personal o impositiva tuya.
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
                "opacity": 1,
                "top": 190,
              }
            }
            testID="ready"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontSize": 17,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                  null,
                  Object {
                    "borderColor": "#fff",
                    "borderRadius": 2,
                    "borderWidth": 1,
                    "color": "#fff",
                    "fontFamily": undefined,
                    "fontSize": undefined,
                    "height": 40,
                    "padding": 10,
                    "textAlign": "center",
                    "width": 247,
                  },
                  null,
                ]
              }
            >
              Listo
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "bottom": 25,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#C4C4C4",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#C4C4C4",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "left": 0,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
        Object {
          "justifyContent": "center",
          "top": 202,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "borderColor": "#fff",
              "borderRadius": 2,
              "borderWidth": 1,
              "color": "#fff",
              "fontFamily": undefined,
              "fontSize": undefined,
              "height": 40,
              "padding": 10,
              "textAlign": "center",
              "width": 247,
            }
          }
        >
          Siguiente
        </Text>
      </View>
    </View>
  </View>
</View>
`;
